### PR TITLE
Set Default EVM to v0.31

### PR DESCRIPTION
**Pull request by shemnon**
**Description**:
Set the default EVM to v0.31, which includes replacing the DIFFICULTY operation with a PRNG operation.

**Related issue(s)**:

#3882 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — DISMISSED** [review]
LGTM

**Review by Neeharika-Sompalli — DISMISSED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Blocking due to failing CI tests

**Review by tinker-michaelj — COMMENTED** [review]
Oh, needs `EVM_VERSION_0_31` [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/hedera-mono-service/src/test/java/com/hedera/services/context/properties/BootstrapPropertiesTest.java#L109).

**Comment by shemnon** [comment]
All the code duplication is in test code.  Can we get that out of the sonar duplication checks?   //WONTFIX

File: test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ private HapiApiSpec contractCallRestrictionsEnforced() {
                                                             asAddress(civilianAccountID.get()),
                                                             asAddress(detachedAccountID.get())
                                                         })
-                                                .hasKnownStatus(INVALID_SOLIDITY_ADDRESS)),
+                                                .hasKnownStatus(FAIL_INVALID)),
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java:127]: Hmm.... this is failing on a transfer...

https://github.com/hashgraph/hedera-services/blob/3f44c6ec2853742971eb95cb5b6879b45c32e2a2/test-clients/src/main/resource/contract/solidity/DoubleSend.sol#L14

I think this is one case we do want to send an invalid solidity error,

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallCodeOperationSuite.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ HapiApiSpec verifiesExistence() {
                 .when()
                 .then(
                         contractCall(contract, "callCode", INVALID_ADDRESS)
-                                .hasKnownStatus(INVALID_SOLIDITY_ADDRESS),
+                                .hasKnownStatus(FAIL_INVALID),
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallCodeOperationSuite.java:54]: Calling a non existent contract w/o balance should either immediately return a true success (as though the code was `0x00` / evm `STOP), not a fail invalid.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallOperationSuite.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ HapiApiSpec verifiesExistence() {
                 .when()
                 .then(
                         contractCall(contract, "call", INVALID_ADDRESS)
-                                .hasKnownStatus(INVALID_SOLIDITY_ADDRESS),
+                                .hasKnownStatus(FAIL_INVALID),
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallOperationSuite.java:65]: This should succeed.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallOperationSuite.java [hunk 4]
```diff
@@ -90,7 +90,7 @@ HapiApiSpec callingContract() {
                         contractCall(contract, "setVar1", 35),
                         contractCallLocal(contract, "getVar1").logged(),
                         contractCall(contract, "callContract", INVALID_ADDRESS, 222)
-                                .hasKnownStatus(INVALID_SOLIDITY_ADDRESS))
+                                .hasKnownStatus(FAIL_INVALID))
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/CallOperationSuite.java:93]: This should succed.

**Review by shemnon — COMMENTED** [review]
Stoping halfway through.  We can't ship this as-is.

**Pull request closed without merging**